opentelemetry-sdk/tests/trace/export/test_export.py

```diff
@@ -148,6 +148,33 @@ def test_flush(self):
 
         span_processor.shutdown()
 
+    def test_multithreading(self):
+        # Should test that shutdown and force_flush can be called concurrently without
+        # causing racing conditions
+
+        spans_names_list = []
+
+        my_exporter = MySpanExporter(destination=spans_names_list)
+        span_processor_to_flush = export.BatchExportSpanProcessor(my_exporter)
+        span_processor_to_shutdown = export.BatchExportSpanProcessor(
+            my_exporter
+        )
+
+        span_names0 = ["xxx", "bar", "foo"]
+        span_names1 = ["yyy", "baz", "fox"]
+
+        for name in span_names0:
+            _create_start_and_end_span(name, span_processor_to_flush)
```

Comment:
Not super familiar with this code, but what happens if spans are "created" by one thread while the other thread is shutting down? Is that something we should test as well?

Reply:
this is a great point. The only reason I didn't look into that in particular is because the two methods outlined in the issue were `force_flush()` and `shutdown()`